src/handlers.rs

```diff
@@ -172,3 +172,72 @@ where
     }
     Ok(())
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    macro_rules! send_to {
+        ($v:expr) => {
+            |bs| {
+                $v.extend_from_slice(bs);
+                Ok(())
+            }
+        };
+    }
+
+    fn test_logger() -> slog::Logger {
+        Logger::root(slog::Discard, slog::o!())
+    }
+
+    #[test]
+    fn test_handle_request_empty_key() {
+        let mut output = vec![];
+        let request = protocol::Request {
+            ty: protocol::RequestType::GETPWBYNAME,
+            key: &[],
+        };
+
+        assert!(
+            handle_request(&test_logger(), &request, send_to!(output)).is_err(),
+            "should error on empty input"
+        );
+        assert!(output.is_empty());
+    }
+
+    #[test]
+    fn test_handle_request_nul_data() {
+        let mut output = vec![];
+        let request = protocol::Request {
+            ty: protocol::RequestType::GETPWBYNAME,
+            key: &[0x7F, 0x0, 0x0, 0x01],
```

Comment:
Is there something special about these bytes?

Reply:
I picked something with an interior null byte just for fun (`127.0.0.1`) but anything that's not a username should work here.